src/plugins/intel_cpu/src/emitters/snippets/riscv64/jit_memory_emitters.cpp

```diff
@@ -0,0 +1,317 @@
+    count = load->get_count();
+    byte_size = src_prc.size();
+}
+
+size_t jit_memory_emitter::aux_gprs_count() const {
+    // for runtime arguments
+    return is_offset_runtime ? 1 : 0;
+}
+
+std::vector<size_t> jit_memory_emitter::get_available_aux_gprs() const {
+    OV_CPU_JIT_EMITTER_ASSERT(IMPLICATION(is_offset_runtime, !aux_gpr_idxs.empty()),
+                              "If offset is dynamic, memory emitter need to have one aux gpr at least!");
+    auto available_aux_gprs = aux_gpr_idxs;
+    if (is_offset_runtime) {
+        available_aux_gprs.pop_back();
+    }
+    return available_aux_gprs;
+}
+
+void jit_memory_emitter::emit_code_impl(const std::vector<size_t>& in_idxs,
+                                        const std::vector<size_t>& out_idxs,
+                                        const std::vector<size_t>& pool_vec_idxs,
+                                        const std::vector<size_t>& pool_gpr_idxs) const {
+    std::vector<size_t> pool_fp_gpr_idxs;  // Empty for now
+    emitter_preamble(in_idxs, out_idxs, pool_vec_idxs, pool_gpr_idxs, pool_fp_gpr_idxs);
+
+    auto reg_runtime_params = Xbyak_riscv::a0;  // First ABI parameter register
+    Xbyak_riscv::Reg aux_gpr = is_offset_runtime ? Xbyak_riscv::Reg(aux_gpr_idxs.back()) : Xbyak_riscv::zero;
+
+    Xbyak_riscv::Reg data_reg = Xbyak_riscv::zero;
+    if (in_out_type_ == emitter_in_out_map::gpr_to_vec) {
+        data_reg = Xbyak_riscv::Reg(in_idxs[0]);
+    } else if (in_out_type_ == emitter_in_out_map::vec_to_gpr) {
+        data_reg = Xbyak_riscv::Reg(out_idxs[0]);
+    } else {
+        OV_CPU_JIT_EMITTER_THROW("unsupported in_out_type");
+    }
+
+    if (is_offset_runtime) {
+        // load the runtime offset from args.buffer_offsets[buffer_cluster_id]
+        const auto offset = GET_OFF(buffer_offsets) + buffer_cluster_id * sizeof(size_t);
+        // RV64 uses 64-bit size_t
+        h->ld(aux_gpr, reg_runtime_params, static_cast<int32_t>(offset));
+        // bump the pointer
+        h->add(data_reg, data_reg, aux_gpr);
+    }
+
+    emit_impl(in_idxs, out_idxs);
+
+    if (is_offset_runtime) {
+        // subtract back so we leave the pointer unchanged for the caller
+        h->sub(data_reg, data_reg, aux_gpr);
+    }
+
+    emitter_postamble();
+}
+
+void jit_load_memory_emitter::emit_impl(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
+    if (host_isa_ == ov::intel_cpu::riscv64::gv) {
+        emit_isa<ov::intel_cpu::riscv64::gv>(in, out);
+    } else {
+        OV_CPU_JIT_EMITTER_THROW("Doesn't support isa ", host_isa_);
+    }
+}
+
+template <cpu_isa_t isa>
+void jit_load_memory_emitter::emit_isa(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
+    auto src = Xbyak_riscv::Reg(in[0]);
+    auto dst = Xbyak_riscv::VReg(out[0]);
+
+    // Set vector configuration for the load (e16 for 2-byte, e32 for 4-byte)
+    auto sew = (byte_size == 2) ? Xbyak_riscv::SEW::e16 : Xbyak_riscv::SEW::e32;
```

Comment:
Should we check that byte_size is equal to 2 or 4? The same for store emitter

Reply:
Added